resources/assets/js/features/wizard-trip/components/StepThree.js

```diff
@@ -1,77 +1,162 @@
 import React from 'react';
-import {Input} from 'app/components/Controls/index.js';
+import {Input, InputAutocomplete} from 'app/components/Controls/index.js';
 import { Button } from 'reactstrap';
 import {connect} from 'react-redux';
 import {bindActionCreators} from 'redux';
 import {addCar} from '../actions';
 import {getTranslate} from 'react-localize-redux';
-import { VehicleValidate } from 'app/services/VehicleService';
+import { VehicleValidate, VehicleData } from 'app/services/VehicleService';
 
 class StepThree extends React.Component {
     constructor() {
         super();
         this.state = {
-            brand: '',
-            model: '',
+            brand: {
+                id: null,
+                value: ''
+            },
+            model: {
+                id: null,
+                value: ''
+            },
+            brands: [],
+            models: [],
             errors: {}
         };
 
-        this.onMarkChange = this.onMarkChange.bind(this);
+        this.onBrandChange = this.onBrandChange.bind(this);
         this.onModelChange = this.onModelChange.bind(this);
         this.onNext = this.onNext.bind(this);
+
+        this.onBrandSelected = this.onBrandSelected.bind(this);
+        this.onModelSelected = this.onModelSelected.bind(this);
+        this.transformBrand = this.transformBrand.bind(this);
+        this.transformModel = this.transformModel.bind(this);
     }
 
     componentWillMount() {
         this.setState({
-            brand: this.props.tripWizard.brand,
-            model: this.props.tripWizard.model
+            brand: {
+                value: this.props.tripWizard.brand
+            },
+            model: {
+                value: this.props.tripWizard.model
+            }
         });
     }
 
-    onMarkChange(e) {
-        this.setState({brand: e.target.value});
+    onBrandChange(e) {
+        const brand = e.target.value,
+            {brands} = this.state;
+
+        this.setState({brand: { value: e.target.value }});
+
+        VehicleData
+            .fetchBrand(brand)
+            .then((response) => this.setState({brands: _.unionBy(brands, response.data.data, 'id')}));
     }
 
     onModelChange(e) {
-        this.setState({model: e.target.value});
+        const model = e.target.value,
+            { brand, models } = this.state;
+
+        this.setState({model: { value: e.target.value}});
+
+        VehicleData
+            .fetchModel(model, brand.id)
+            .then((response) => this.setState({models: _.unionBy(models, response.data.data, 'id')}));
```

Comment:
don't store all models that was fetched, just enough to store last fetched collection